libs/blocks/editorial-card/editorial-card.js

```diff
@@ -0,0 +1,121 @@
+import { createTag, loadStyle, getConfig } from '../../utils/utils.js';
+import { decorateBlockBg, decorateBlockText, decorateBlockHrs, decorateTextOverrides, applyHoverPlay } from '../../utils/decorate.js';
+
+function decorateLockupFromContent(el) {
+  const rows = el.querySelectorAll(':scope > div > p');
+  const firstRowImg = rows[0]?.querySelector('img');
+  if (firstRowImg) {
+    rows[0].classList.add('lockup-area');
+    rows[0].childNodes.forEach((node) => {
+      if (node.nodeType === 3 && node.nodeValue !== ' ') {
+        const newSpan = createTag('span', { class: 'lockup-label' }, node.nodeValue);
+        node.parentElement.replaceChild(newSpan, node);
+      }
+    });
+  }
+}
+
+const extendDeviceContent = (el) => {
+  // check if there is a .body- above the .detail-
+  const detail = el.querySelector('[class^="detail-"]');
+  const prevElem = detail?.previousElementSibling;
+  if (prevElem) {
+    const prevIsBody = Array.from(prevElem.classList).some((c) => c.startsWith('body-'));
+    if (!prevIsBody) return;
+    prevElem.classList.remove('body-m');
+    prevElem.classList.add('body-xxs', 'device');
+  }
+};
```

Comment:
You can remove the nested if's
```suggestion
const extendDeviceContent = (el) => {
  const detail = el.querySelector('[class^="detail-"]');
  const prevElem = detail?.previousElementSibling;
  if (!prevElem || ![...prevElem.classList].some(c => c.startsWith('body-'))) return
  prevElem.classList.remove('body-m');
  prevElem.classList.add('body-xxs', 'device');
};
```